parsers/verilog.c

```diff
@@ -1577,8 +1580,23 @@ static int tagNameList (tokenInfo* token, int c)
 	{
 		c = processType(token, c, &kind);	// update token and kind
 
-		if (c == '=' || c == ',' || c == ';' || c == ')' || c == '`')
+		if (c == '=' || c == ',' || c == ';' || c == ')' || c == '`' || c == '(')
 		{
+			if(c == '(')
+			{
+				token->kind = K_INSTANCE;
+				kind = K_INSTANCE;
+				c = skipPastMatch ("()");
+				if(c != ';' && c != ',')
```

Comment:
We can simplify the code a bit like:
```
if (c == '(')
{
      kind = K_UNDEFINED;
      c = skipPastMatch ("()");
      if (c == ';' || c == ',')
      {
			kind = K_INSTANCE;
            verbose("find instance: %s with kind %s\n", vStringValue (token->name), getNameForKind(kind));        
      }
      token->kind = kind;
}
```

Reply:
I will check this code in detail in this week.
Quick comments.

I propose the followings; 

```
if (c == '(')
{
      c = skipPastMatch ("()");
      if (c == ';' || c == ',')
      {
		kind = K_INSTANCE;
        	verbose("find instance: %s with kind %s\n", vStringValue (token->name), getNameForKind(kind));        
      }
      else  // comment why this condition is required
		kind = K_UNDEFINED;

      token->kind = kind;
}
```

Actually I don't understand why the condition is required. 

From LRM 23.3.2 Module instantiation syntax

```
module_instantiation ::= // from A.4.1.1
	module_identifier [ parameter_value_assignment ] hierarchical_instance { , hierarchical_instance };
...
hierarchical_instance ::= name_of_instance ( [ list_of_port_connections ] )
name_of_instance ::= instance_identifier { unpacked_dimension }
...
```

Please take care of ` { unpacked_dimension }` part.

Please check interface instantiation syntax, too.

Reply:
From the units test, the parser will treat "`add_t" as instance if there isn't a "else" condition
``` verilog
    var `add_t(foo) = '0;
```
I will check unpacked_dimension and interface syntax later, thanks.